src/SimpleEndPoints/Conventions/EndpointRoutingConvention.cs

```diff
@@ -1,22 +1,77 @@
-﻿using Microsoft.AspNetCore.Mvc.ApplicationModels;
+﻿using System;
+using System.Linq;
+using System.Text;
+using Microsoft.AspNetCore.Mvc.ApplicationModels;
+using Microsoft.AspNetCore.Routing;
+using SimpleEndpoints.Extensions;
+using SimpleEndpoints.VerbScoped;
 
 namespace SimpleEndpoints.Conventions
 {
-    public class EndpointRoutingConvention: IApplicationModelConvention
+    public class EndpointRoutingConvention : IApplicationModelConvention
     {
-        private const string EndpointString = "endpoint";
+        private const string EndpointPlaceholder = "[endpoint]";
+
+        private readonly SimpleEndpointsConfiguration _configuration;
+
+        public EndpointRoutingConvention(SimpleEndpointsConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
 
         public void Apply(ApplicationModel application)
         {
             foreach (var controller in application.Controllers)
             {
-                controller.Selectors[0].AttributeRouteModel.Template = controller
-                    .Selectors[0]
-                    .AttributeRouteModel
-                    .Template
-                    .Replace($"[{EndpointString}]", 
-                        controller.ControllerName.Substring(0,
-                        controller.ControllerName.Length - EndpointString.Length));
+                MapRouteConvention(controller);
+            }
+        }
+
+        private  void MapRouteConvention(ControllerModel controller)
+        {
+            var routeTemplate = controller.Selectors[0].AttributeRouteModel.Template;
+
+            //If we haven't set a route with [Route("")] on our endpoint
+            if (routeTemplate.Equals(EndpointPlaceholder, StringComparison.OrdinalIgnoreCase))
```

Comment:
Should this not be contains() to cater for `[Route("custom/[endpoint]/another")]`

Reply:
It did occur to me that this was your intention originally and i had removed it without knowing, a cool feature that I will add back in.